Transaction tests assume database schema app

The relevant specification version and section number(s): 2.1

The coordinates of the challenged test(s): com.ibm.jbatch.tck.tests.jslxml.TransactionTests

The exact TCK version: 2.1

The implementation being tested, including name and company: JBeret, by Red Hat, Inc.

A full description of why the test is invalid and what the correct behavior is believed to be:
Any supporting material; debug logs, test output, test logs, run scripts, etc.:

`TransactionTests` assumes all database tables are in `APP` schema. This is the default schema in derby. So when testing with Derby, it always works, with this schema name in some places, and without it in other places. But it will not work with any other db.

The [ConnectionHelper](https://github.com/eclipse-ee4j/batch-tck/blob/master/com.ibm.jbatch.tck/src/main/java/com/ibm/jbatch/tck/artifacts/chunkartifacts/ConnectionHelper.java#L40) class and RetryConnectionHelper class use `app` schema in all sql statements.

[DDL files](https://github.com/eclipse-ee4j/batch-tck/blob/master/com.ibm.jbatch.tck/src/main/resources/ddls/postgresql.ddl.jbatch-tck.sql) don't use schema.

Failed tests:
```
[ERROR] Failures:
[ERROR]   TransactionTests.testGlobalTranForcedExceptionCheckpointRestart:349 Expected value: Inventory=81 InitialCheckpoint=null OrderCount=18, but found value: FAILED
[ERROR]   TransactionTests.testGlobalTranForcedExceptionWithRollback:295 Expected value: Inventory=81 InitialCheckpoint=null OrderCount=18, but found value: FAILED
[ERROR]   TransactionTests.testGlobalTranLongDelayMixOfLongTimeoutStepsAndShortTimeoutSteps:477 Expected value: COMPLETED, but found value: FAILED
[ERROR]   TransactionTests.testGlobalTranLongDelayMixOfLongTimeoutStepsAndShortTimeoutStepsCustomCheckpointAlgorithm:550 Expected value: COMPLETED, but found value: FAILED
[ERROR]   TransactionTests.testGlobalTranNoDelayLongTimeout:416 Expected value: Inventory=0 InitialCheckpoint=null OrderCount=99, but found value: FAILED
[ERROR]   TransactionTests.testGlobalTranNoExceptions:243 Expected value: Inventory=0 InitialCheckpoint=null OrderCount=99, but found value: FAILED
[ERROR]   TransactionTests.testTranRollbackRetryProcessSkipProcess:146 Expected value: COMPLETED, but found value: FAILED
[ERROR]   TransactionTests.testTranRollbackRetryReadSkipRead:99 Expected value: COMPLETED, but found value: FAILED
[ERROR]   TransactionTests.testTranRollbackRetryWriteSkipWrite:193 Expected value: COMPLETED, but found value: FAILED
``` 

I suggest removing schema name from all sql statements, to make it easy to configure with other db products.

--- a/jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc
+++ b/jakarta.batch.official.tck/src/main/asciidoc/batch-tck-reference-guide.adoc
@@ -416,7 +416,21 @@ The test suite files, the test database DDLs, and reporting XSL can be extracted
 
 ==== REQUIRED - test database setup - DDL files
 
-The test suite includes tests that use an application database which must be created separately, prior to TCK execution, potentially in an application-specific way.  The DDL defining the required table format is specified in a set of .ddl files included in the 'com.ibm.jbatch.tck' module dependency.  These can most conveniently be accessed by adding the dependency plugin with parent configuration, as mentioned in the previous section.   They can also be accessed in the TCK zip at file: `artifacts/com.ibm.jbatch.tck-<version>.jar`. 
+The test suite includes tests that use an application database which must be created separately, prior to TCK execution, potentially in an implementation-specific way.  The DDL defining the required table format is specified in a set of *.sql files included in the 'com.ibm.jbatch.tck' module dependency.  These can most conveniently be accessed by adding the dependency plugin with parent configuration, as mentioned in the previous section.   They can also be accessed in the TCK zip at file: `artifacts/com.ibm.jbatch.tck-<version>.jar`. 
+
+==== NOTE: database setup process may require extra steps for non-Derby databases
+
+During TCK execution, the test application accesses database tables with the 'app' high-level qualifier (or "schema"), e.g. `insert into app.numbers values(?, ?)`.  However, this 'app' high-level qualifier does not appear in the *.sql setup DDL files provided by the TCK.  These DDL files reference tables without a high-level qualifier.
+
+Since the Apache Derby database uses 'app' as the default high-level qualifier (schema), the runtime access aligns with the setup DDL when Derby is used as the database product.  
+
+However, when setting up other databases not using 'app' as the default schema, additional step(s) may be needed.  This could involve modifying the setup DDL to set 'app' as the high-level qualifier and/or setting the current schema way when executing the database setup DDL, perhaps using a database-specific tool.
+
+In issue:  https://github.com/eclipse-ee4j/batch-tck/issues/51 a challenge was raised because the TCK seemed too dependent on the use of the Derby database.
+
+The challenge was accepted but it was agreed to resolve this for the current TCK release by documenting the workaround (in the documentation section you are reading now) and open an issue to enhance the TCK in a future release.  This future enhancement could either allow the schema to be parameterized or perhaps removed altogether.
+
+==== NOTE: any database with JDBC driver may be used
 
 There is no requirement to use one of the databases for which a DDL is included; another database with JDBC-compliant driver could potentially be used.
 

